lib/Doctrine/Migrations/Generator/DiffGenerator.php

```diff
@@ -95,15 +96,27 @@ static function ($assetName) use ($filterExpression) {
 
         $toSchema = $this->createToSchema();
 
+        if (method_exists($this->schemaManager, 'createComparator')) {
+            $comparator = $this->schemaManager->createComparator();
+        }
+
+        $upSql = isset($comparator) ?
+            $comparator->compareSchemas($fromSchema, $toSchema)->toSql($this->platform) :
+            $fromSchema->getMigrateToSql($toSchema, $this->platform);
```

Comment:
At first, I tried inlining the contents of this method to avoid the deprecation, but then I could no longer mock anything.